ci/terraform/shared/iam.tf

```diff
@@ -2,7 +2,7 @@
 
 # Create a new IAM role for cross account access
 resource "aws_iam_role" "cross_account_role" {
-  count       = var.environment == "build" ? 1 : 0
+  count       = var.environment == "build" || var.environment == "staging" ? 1 : 0
   name        = "CrossAccountRole-new-build"
```

Comment:
Should the name have environment variable on it? i.e.
```
  name        = "CrossAccountRole-new-${var.environment}"
```

Reply:
Additionally, the assume_role_policy also require Staging account id in the statement